pkg/plugins/codegen/jenny_plugintstypes.go

```diff
@@ -0,0 +1,50 @@
+package codegen
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/grafana/pkg/plugins/codegen/kindsys"
+)
+
+func PluginTSTypesJenny(root string, inner codejen.OneToOne[*kindsys.PluginDecl]) codejen.OneToOne[*kindsys.PluginDecl] {
+	return &ptsJenny{
+		root:  root,
+		inner: inner,
+	}
+}
+
+type ptsJenny struct {
+	fileName string
+	root     string
+	inner    codejen.OneToOne[*kindsys.PluginDecl]
+}
+
+func (j *ptsJenny) JennyName() string {
+	return "PluginTSTypesJenny"
+}
+
+func (j *ptsJenny) Generate(decl *kindsys.PluginDecl) (*codejen.File, error) {
+	tf := templateVars_autogen_header{
+		GeneratorPath: "public/app/plugins/gen.go", // FIXME hardcoding is not OK
+		LineagePath:   "models.cue",                // FIXME hardcosing is not OK
+	}
+
+	var buf bytes.Buffer
+	err := tmpls.Lookup("autogen_header.tmpl").Execute(&buf, tf)
+	if err != nil {
+		return nil, fmt.Errorf("error executing header template: %w", err)
+	}
```

Comment:
This approach to header injection does work, and has the (huge) advantage of having access to the jenny input, which makes it feasible to indicate the original input in the generated header.

What we do with core kinds is [use a `FileMapper` as a `JennyList` postprocessor](https://github.com/grafana/grafana/blob/main/kinds/gen.go#L46), which is considerably simpler to implement.

no need to switch, just food for thought

Reply:
Ah - that was a great suggestion! Will do something similar in this code! Thank 🙏🏻